hydro_serving_writer/buffer.py

```diff
@@ -0,0 +1,163 @@
+import os, logging, datetime
+from typing import Dict
+from collections import defaultdict
+from concurrent import futures
+import pyarrow as pa, pyarrow.parquet as pq
+import google
+from hydro_serving_grpc import writer
+from hydro_serving_writer.storage import Storage
+from hydro_serving_writer.utils.config import get_config
+
+__all__ = ["ModelVersionBufferPool"]
+
+logging.basicConfig(level=logging.DEBUG)
+logging.getLogger('matplotlib').setLevel(logging.WARNING) 
+logger = logging.getLogger()
+
+scheme = pa.schema([
+    ('model_version', pa.int64()), 
+    ('feature', pa.string()),
+    ('check', pa.bool_()),
+    ('timestamp', pa.timestamp('ms')),
+    ('uid', pa.int64()),
+    ('value', pa.float64()),
+    ('description', pa.string())
+])
+
+
+class ModelVersionBufferPool:
+    pool = {}
+    config = get_config()
+
+    @classmethod
+    def get_buffer(cls, model_version):
+        if ModelVersionBufferPool.pool.get(model_version):
+            return ModelVersionBufferPool.pool.get(model_version)
+        ModelVersionBufferPool.pool[model_version] = ModelVersionBuffer(
+            model_version, ModelVersionBufferPool.config.get("BUFFER_SIZE"))
+        return ModelVersionBufferPool.pool[model_version]
+    
+    @staticmethod
+    def write_one(request: writer.WriteOneRequest):
+        buffer = ModelVersionBufferPool.get_buffer(request.model_version)
+        buffer.write(request.ByteSize(), request.SerializeToString())
+
+    @staticmethod
+    def write_many(request: writer.WriteManyRequest):
+        for model_version, request in ModelVersionBufferPool.group_request(request).items():
+            buffer = ModelVersionBufferPool.get_buffer(model_version)
+            buffer.write(request.ByteSize(), request.SerializeToString())
+
+    @staticmethod
+    def group_request(request: writer.WriteManyRequest) -> Dict[int, writer.WriteManyRequest]:
+        group = defaultdict(writer.WriteManyRequest)
+        for row in zip(*(getattr(request, name) for name in scheme.names)):
+            group[row[0]].MergeFrom(
+                writer.WriteManyRequest(
+                    **dict(zip(scheme.names, map(lambda x: [x], row)))
```

Comment:
I think that using a dict comprehension makes the expression less complex.
```suggestion
                    **{name: [value] for name, value in zip(scheme.names, values)}
```

Reply:
Yeah, looks better, thank you